.github/workflows/openhands-resolver.yml

```diff
@@ -253,31 +253,57 @@ jobs:
               grep "branch created" branch_result.txt | sed 's/.*\///g; s/.expand=1//g' > branch_name.txt
           fi
 
-      - name: Comment on issue
+      # Step leaves comment for when agent is invoked on PR
+      - name: Analyze Push Logs (Updated PR or No Changes) # Skip comment if PR update was successful OR leave comment is agent made no code changes
         uses: actions/github-script@v7
-        if: always() # Comment on issue even if the previous steps fail
+        if: always()
+        env:
+          CONDITIONAL_EXECUTED: ${{ env.CONDITIONAL_EXECUTED || 'false' }}
         with:
           github-token: ${{ secrets.PAT_TOKEN || github.token }}
           script: |
             const fs = require('fs');
             const issueNumber = ${{ env.ISSUE_NUMBER }};
-            const success = ${{ steps.check_result.outputs.RESOLUTION_SUCCESS }};
-
-            let prNumber = '';
-            let branchName = '';
             let logContent = '';
-            const noChangesMessage = `No changes to commit for issue #${issueNumber}. Skipping commit.`;
 
             try {
-              if (success){
-                logContent = fs.readFileSync('/tmp/pr_result.txt', 'utf8').trim();
-              } else {
-                logContent = fs.readFileSync('/tmp/branch_result.txt', 'utf8').trim();
-              }
+              logContent = fs.readFileSync('/tmp/pr_result.txt', 'utf8').trim();
             } catch (error) {
-              console.error('Error reading results file:', error);
+              console.error('Error reading pr_result.txt file:', error);
+            }
+
+            const noChangesMessage = `No changes to commit for issue #${issueNumber}. Skipping commit.`;
+
+            // Check logs from send_pull_request.py (pushes code to GitHub)
+            if (logContent.includes("Updated pull request")) {
+              console.log("Updated pull request found. Skipping comment.");
+              process.env.CONDITIONAL_EXECUTED = 'true';
+            } else if (logContent.includes(noChangesMessage)) {
+              github.rest.issues.createComment({
+                issue_number: issueNumber,
+                owner: context.repo.owner,
+                repo: context.repo.repo,
+                body: `The workflow to fix this issue encountered an error. Openhands failed to create any code changes.`
```

Comment:
Not a review comment, just a funny fyi note: actually this isn't _necessarily_ an error either. I have used the resolver a couple of times with a prompt like "if [something], then stop there and add all this information in your finish action."
So it wasn't, strictly speaking, an error when the LLM did just that: no code changes, and tell me why. 😇

(I'm not suggesting to change this though, idk)